### PR TITLE
ZCS-5071 - setup dialog screens with basic controls

### PR DESCRIPTION
**Pull request by naijenn**
Wrote out the dialog screens with basic controls for password reset functionality from login screen. No connection to password reset apis is connected as it is not a part of this acceptance criteria. Button names will be changed in future work. For example the 'Next' button steps through the process, but at certain points will have the wording of the button changed to fit the need of the view at the time. The 'Previous' button will be used in actual flow only when it should be available. Additional css required on future tickets as well.

### CODE REVIEW

File: WebRoot/js/zimbraMail/package/PasswordRecovery.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ *
+ * Zimbra Collaboration Suite Web Client
+ * Copyright (C) 2015, 2016 Synacor, Inc.
```
Comment (shrikant-prasad) [inline WebRoot/js/zimbraMail/package/PasswordRecovery.js]: Please update the year to 2018 here and in other new files copyright section.

File: WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+ *
+ * All portions of the code are Copyright (C) 2015, 2016, 2017, 2018 Synacor, Inc. All Rights Reserved.
+ * ***** END LICENSE BLOCK *****
+ */
+
+/**
+ * Creates a dialog for Two factor initial setup
+ * @constructor
+ * @class
+ * @author  Hem Aravind
```
Comment (shrikant-prasad) [inline WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js]: update author name.

File: WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+ * The Original Code is Zimbra Open Source Web Client.
+ * The Initial Developer of the Original Code is Zimbra, Inc.  All rights to the Original Code were
+ * transferred by Zimbra, Inc. to Synacor, Inc. on September 14, 2015.
+ *
+ * All portions of the code are Copyright (C) 2015, 2016, 2017, 2018 Synacor, Inc. All Rights Reserved.
+ * ***** END LICENSE BLOCK *****
+ */
+
+/**
+ * Creates a dialog for Two factor initial setup
```
Comment (shrikant-prasad) [inline WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js]: incorrect description

File: WebRoot/messages/ZmMsg.properties [hunk 4]
```diff
@@ -4450,7 +4451,7 @@ QUERY_IS_unread = unread
 QUERY_IS_unreplied = not replied to by me
 
 # Two step authentication setting dialog
-twoStepAuthSetup = Set up two-step authentication
+twoStepAuthSetup = Set up TEN-step authentication
```
Comment (shrikant-prasad) [inline WebRoot/messages/ZmMsg.properties]: it should be two-step.

File: WebRoot/public/PasswordRecovery.jsp [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+		<jsp:param name="res" value="I18nMsg,AjxMsg,ZmMsg" />
+		<jsp:param name="skin" value="${skin}" />
+	</jsp:include>
+
+	<jsp:include page="Boot.jsp"/>
+
+	<% if (isDebugMode != null) { %>
+		<jsp:include page="/public/jsp/PasswordRecovery.jsp"/>
+	<% } else { %>
+		<script src="${contextPath}/js/PasswordRecovery_all.js<%=ext%>?v=${version}"></script>
```
Comment (shrikant-prasad) [inline WebRoot/public/PasswordRecovery.jsp]: there isn't any PasswordRecovery_all.js

File: WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js (hunk 2 above)
Comment (chinmay15) [reply WebRoot/js/zimbraMail/share/view/dialog/ZmPasswordRecoveryDialog.js]: I'll suggest to remove @author, we generally don't keep this documented.

File: WebRoot/public/PasswordRecovery.jsp (hunk 5 above)
Comment (naijenn) [reply WebRoot/public/PasswordRecovery.jsp]: PasswordRecovery_all and other files are generated in build/WebRoot/js directory through ant prod-deploy

**Comment by naijenn** [comment]
This ticket was to build the dialogs and get basic controls in place. As for Two factor, some discussion is required and is a bit out of scope of this ticket.

File: WebRoot/public/login.jsp [hunk 6]
```diff
@@ -584,6 +584,10 @@ if (application.getInitParameter("offlineMode") != null) {
                                     <input id="remember" value="1" type="checkbox" name="zrememberme" />
                                     <label for="remember"><fmt:message key="${smallScreen?'rememberMeMobile':'rememberMe'}"/></label>
                                 </c:if>
+                                <script TYPE="text/javascript">
```
Comment (silentsakky) [inline WebRoot/public/login.jsp]: any specific reason to use script tag? we can directly add html code

File: WebRoot/public/PasswordRecovery.jsp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	<% } %>
+</head>
+<body class="user_font_system">
+	<script>
+		if(!window.DBG) {
+			DBG = new AjxDebug(AjxDebug.NONE, null, false);
+		}
+		var params = {
+			accountInput : "<%=accountInput%>",
+			isFromLoginPage : true
```
Comment (silentsakky) [inline WebRoot/public/PasswordRecovery.jsp]: why do we need this flag? are we going to open this dialog from somewhere else also?

**Pull request merged**